app/src/main/java/com/toshi/service/GcmMessageReceiver.java

```
@@ -155,10 +155,7 @@ private void handleInvalidPayment(final Throwable throwable) {
     private void tryShowIncomingMessage() {
         final IncomingMessage incomingMessage;
         try {
-            incomingMessage = BaseApplication
-                .get()
-                .getSofaMessageManager()
-                .fetchLatestMessage();
+            incomingMessage = getIncomingMessage();
         } catch (final TimeoutException e) {
```

`TimeoutException` is no longer thrown so this can be removed.

---

Yes it's. `SofaMessageManager::fetchLatestMessage` throws `TimeoutException`

---

ok, so `onErrorResumeNext` doesn't actually swallow the exception -- it's still being thrown?

---

`GcmMessageReceiver` calls `SofaMessageManager::fetchLatestMessage`, not `SofaMessageReceiver::fetchLatestMessage` 👇 

```
public Single<IncomingMessage> fetchLatestMessage() throws TimeoutException {
        try {
            while (this.messageReceiver == null) {
                Thread.sleep(200);
            }
        } catch (final InterruptedException e) {
            throw new TimeoutException(e.toString());
        }
        return this.messageReceiver.fetchLatestMessage();
    }
```

---

Oh right, there's no reason for that to throw `TimeoutException` now since message receiver no longer throws `TimeoutException` -- it can just throw `InterruptedException`